tests/interfaces/test_autograd.py

```diff
@@ -475,3 +475,234 @@ def d_classical(a, b, method):
         assert grad_fd1 == pytest.approx(grad_angle, abs=tol)
         assert grad_fd1 == pytest.approx(grad_auto, abs=tol)
         assert grad_angle == pytest.approx(grad_auto, abs=tol)
+
+
+class TestParameterHandlingIntegration:
+    """Test that the parameter handling for differentiable/non-differentiable
+    parameters works correctly."""
+
+    def test_differentiable_parameter_first(self, mocker):
+        """Test that a differentiable parameter used as the first
+        argument is correctly evaluated by QNode.jacobian, and that
+        all other non-differentiable parameters are ignored"""
+        spy = mocker.spy(qml.qnodes.JacobianQNode, "jacobian")
```

Comment:
:shipit:

Reply:
Is this a new feature? Haven't seen it used before

Reply:
Kind of, but also not really 😆 

`mocker` is simply a pytest fixture that 'wraps' `unittest.mock`, providing a more convenient UI. So this is doable using `unittest.mock`/subclasses/`with` contexts, but using this fixture makes it significantly more readable I feel.